### DISCUSSION TITLE
多线程访问同一个资源，限流策略不生效

### DISCUSSION THREAD

**Discussion by kangzhixing** [alibaba/Sentinel#2423] [Q&A] [answered]
使用sentinel过程中测试发现如下问题：
定义了一个资源，qps为2，预期为多线程下，该资源1秒内仅仅能被访问两次。
但是，测试结果与预期不同

1. 如果同一个线程不断访问某个资源100次，限流生效，成功访问两次；
2. 如果开启100个线程并发访问某一个资源，每个线程访问一次，限流不生效；
3. 如果开启5个线程，每个线程访问20次，则成功访问10次。

测试结果显示限流策略生效于线程内，请问是否是设计如此？

**Comment by kangzhixing** [comment, marked as answer]
已经解决：sentinel初始化过程非线程安全，并发测试前先同步访问资源一次可避免上述问题

**Reply by sczyh30** [reply]
This can be improved. Contributions are welcomed.
